resources/gcp/environment.go

```diff
@@ -77,12 +77,13 @@ func logIn(ctx *pulumi.Context) {
 		}
 	}
 
-	// Environment variable provided in the CI, to activate the service-account authentication
-	if os.Getenv("GOOGLE_CREDENTIALS_FILE") != "" {
-		fmt.Println("GOOGLE_CREDENTIALS_FILE environment detected, activating service account authentication")
-		cmd := exec.Command("gcloud", "auth", "activate-service-account", "--key-file", os.Getenv("GOOGLE_CREDENTIALS_FILE"))
-		if err := cmd.Run(); err != nil {
-			ctx.Log.Error(fmt.Sprintf("Error running `gcloud auth activate-service-account --key-file $GOOGLE_CREDENTIALS_FILE`: %v", err), nil)
+	// Passing the environment variable is not enough for the gcloud CLI to work, per https://cloud.google.com/docs/authentication/provide-credentials-adc
+	// We need the CLI to be working for accessing GKE cluster with Pulumi Kubernetes provider.
+	if path, ok := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS"); ok {
+		cmd := exec.Command("gcloud", "auth", "activate-service-account", "--key-file", path)
```

Comment:
valid comment from the security both, maybe using https://pkg.go.dev/path#Clean

Reply:
Actually if `GOOGLE_APPLICATION_CREDENTIALS` is set there is no need to pass it, `gcloud` should read it and use it.

Reply:
Actually, it looks like `gcloud` does not automatically picks up `GOOGLE_APPLICATION_CREDENTIALS` (Google SDK does tho).
So we need to explicitly call `gcloud` with the credentials file: https://serverfault.com/questions/848580/how-to-use-google-application-credentials-with-gcloud-on-a-server

Reply:
Regarding the security bot, I think this is not a real threat. I use `exec.Command` to pass the args one by one, as indiciated in
>4. Avoid String Concatenation: Refrain from dynamically constructing commands by concatenating strings. Instead, use the arg ...string parameter of exec.Command to pass arguments safely.
So I think this is safe in this case. It would have been an issue with string concatenation indeed

Reply:
The security problem is you are passing the output of `os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS")` as argument of `--key-file`. If you don't sanitise `path`, the risk is to pass for instance 

`GOOGLE_APPLICATION_PATH="something; echo /root/passwords"` and ending up in:

```
gcloud auth activate-service-account --key-file something; echo /root/passwords
```

Reply:
Tested it with a tiny `main.go`

```go
	cmdArgs := "ls; echo 'Hello, random command!'"
	out, err := exec.Command("sh", "-c", cmdArgs).Output()
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("The date is %s\n", out)
```

And I get

```bash
....
Hello, random command!
```

Reply:
Validated that this works only for a subset of commands, and with gcloud the argument path is passed as a single string, so it would fail if it contained injected commands